vk_video_decoder/libs/VkShell/Shell.cpp

```diff
@@ -304,9 +304,9 @@ void Shell::AcquireBackBuffer(bool trainFrame) {
     // acquire just once when not presenting
     if (m_settings.noPresent && GetCurrentBackBuffer().GetAcquireSemaphore() != VK_NULL_HANDLE) return;
 
-    AcquireBuffer* acquireBuf = m_ctx.acquireBuffers.front();
+    assert(!m_ctx.acquireBuffers.empty());
```

Comment:
I don't understand why do we need this change?

The rest of the changes make sense and I'll push alternative changes for those.

Reply:
`assert(acquireBuf != nullptr);` depends on `front()` returning a nullptr when empty, but this is undefined behavior and in practice it does not return a nullptr with libstdc++.

This replaces that assert with an assert that acquireBuffers is `!empty()`

The diff maybe looks a bit odd since the assert is moved before the buffer the front() call.